### PR TITLE
Lots of changes

### PR DESCRIPTION
**Pull request by goosemo**
Need to pin the docker images as well as add a dep that was upgraded out
from under python's docker image.

I also had to specify real users for the SLUMBER work, and the real
hostname for the the env vars.

I also fixed up the docker image for rtd to have the tex work needed to
build pdfs, since that's on be default, and I was tired of it failing
builds on creation.

I also upgraded our RTD code to the latest master, since there were a
bunch of transative deps that were becoming problematic to use
unpinned/unfrozen.

[ticket: DEVOPS-969]

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,23 +1,30 @@
...
+        texlive-latex-recommended \
+        texlive-fonts-recommended \
+    && apt-get -y autoremove \
+    && apt-get clean
+
+WORKDIR /usr/src/app/readthedocs.org
 
 ## Apply our own overrides
-#COPY local_settings.py /usr/src/app/readthedocs.org/readthedocs/settings/local_settings.py
+COPY local_settings.py.example /usr/src/app/readthedocs.org/readthedocs/settings/local_settings.py
```
Comment (jpalawaga) [inline Dockerfile:21]: Are you sure that you want to copy the `.example` file onto `local_settings.py`?

File: Dockerfile [hunk 2]
```diff
@@ -1,23 +1,30 @@
...
+COPY entrypoint.sh ./
 
-RUN cp ./local_settings.py readthedocs.org/readthedocs/settings/local_settings.py
+RUN pip install setuptools==19.2
```
Comment (jpalawaga) [inline Dockerfile:24]: If we've upgraded to rtd latest, do we still need this?

btw, since this submodule points at a specific commit, should we just pip freeze and use that as a requirements file?

File: docker-compose.yml [hunk 3]
```diff
@@ -4,8 +4,8 @@ rtd:
     volumes:
         - ./db:/persistent
     environment:
-        - "TEST_DATA=yes"
-        - "RTD_PRODUCTION_DOMAIN=production-domain:80"
+        - "TEST_DATA=no"
+        - "RTD_PRODUCTION_DOMAIN=docs.planfront.net:80"
```
Comment (jpalawaga) [inline docker-compose.yml:8]: only lame thing about this change is we can't contribute the changes back to the parent repo (if we care about doing so)

File: Dockerfile [hunk 4]
```diff
@@ -1,23 +1,30 @@
...
+        texlive-fonts-recommended \
+    && apt-get -y autoremove \
+    && apt-get clean
+
+WORKDIR /usr/src/app/readthedocs.org
 
 ## Apply our own overrides
-#COPY local_settings.py /usr/src/app/readthedocs.org/readthedocs/settings/local_settings.py
+COPY local_settings.py.example /usr/src/app/readthedocs.org/readthedocs/settings/local_settings.py
+COPY entrypoint.sh ./
```
Comment (jpalawaga) [inline Dockerfile:22]: pardon my docker ignorance, but why is this needed now, but not before?

**Comment by owenstone** [comment]
Thanks for walking me through the changes. Looks good!

**Review by owenstone — APPROVED** [review]
LGTM

File: Dockerfile (hunk 1 above)
Comment (goosemo) [reply Dockerfile:21]: I've edited it to be ours, and the gitignore makes it not track plain `local_settings.py`. I'm basically just making this repo our own.

File: docker-compose.yml (hunk 3 above)
Comment (goosemo) [reply docker-compose.yml:8]: I do not :|, also we could just make this a real env var, and have this be a part of the run command for `docker-compose` and make that upstream change

File: Dockerfile (hunk 2 above)
Comment (goosemo) [reply Dockerfile:24]: yeah it's still required, because the python docker image messes this up for us

File: Dockerfile (hunk 4 above)
Comment (goosemo) [reply Dockerfile:22]: Because I stopped using the `-onbuild` version which does work at build time that's implicit, and essentially hidden to the user. 

One step it did was to `# 3) COPY . /usr/src/app` so now we have to do this manually. I also am doing this in specific steps to keep a good working docker cache so rebuilds are quick.
